payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherActivity.kt

```diff
@@ -80,73 +86,57 @@ internal class GooglePayLauncherActivity : AppCompatActivity() {
         fadeOut()
     }
 
-    private fun payWithGoogle(task: Task<PaymentData>) {
-        AutoResolveHelper.resolveTask(
-            task,
-            this,
-            LOAD_PAYMENT_DATA_REQUEST_CODE
-        )
-    }
+    private fun onGooglePayResult(taskResult: ApiTaskResult<PaymentData>) {
+        when (taskResult.status.statusCode) {
+            CommonStatusCodes.SUCCESS -> {
+                onGooglePayResult(taskResult.result!!)
```

Comment:
The object is guaranteed to be defined by contract at this point.
Expl: the library only contains Java code ATM. Until the API adds Kotlin code, an extension sealed/abstract class can be added on the Kotlin end.